test/service/test_request_validation.py

```diff
@@ -265,3 +271,17 @@ def test(url, message, params):
             test(url,
                  params={},
                  message='Missing required query parameters `entity_type`, `integration_type`')
+
+    def test_bad_catalog_param(self):
+        url = furl(url=self.base_url, path='/index/files').url
+        for invalid_catalog_name, error_message in [
+            ('foo', f"BadRequestError: Catalog name 'foo' is invalid. Must be one of "
```

Comment:
To simplify this and make it less brittle lets just assert that the error message contains `Catalog name 'foo' is invalid` using assertIn